lib/Sema/TypeCheckStmt.cpp

```diff
@@ -767,7 +767,8 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
       // statement, produce a more specific error.
       if (S->getTargetName().empty() && !ActiveLabeledStmts.empty() &&
           (isa<IfStmt>(ActiveLabeledStmts.back()) ||
-           isa<DoStmt>(ActiveLabeledStmts.back())))
+           isa<DoStmt>(ActiveLabeledStmts.back()) ||
+           isa<GuardStmt>(ActiveLabeledStmts.back())))
```

Comment:
Does this work for this?
```swift
func fn(a: Int) {
  guard a < 1 else {
    break
  }
}
```
In this case, IMO, the message should be normal `'break' is only allowed inside a loop, if, do, or switch`.